pkg/endpoint/bpf.go

```diff
@@ -748,6 +741,13 @@ func (e *Endpoint) runPreCompilationSteps(regenContext *regenerationContext, rul
 
 	// Set the computed policy as the "incoming" policy. This can fail if
 	// the endpoint's security identity changed during or after policy calculation.
+	// Note that the incoming policy can be the same as the previous policy in cases
+	// where an unnecessary policy computation was skipped. In that case
+	// e.desiredPolicy == e.realizedPolicy also after this call.
+	// TODO: Figure out how much of the following logic would be safe to skip if there
+	// is no new incoming policy. Policy update can be triggered for changes in named ports
+	// and Envoy listeners, so it seems policy maps and Envoy network policy will need to
+	// be recomputed regardless.
```

Comment:
Do you have plans to follow up on this TODO? Typically TODOs that are added to a codebase just get ignored and become stale, so we tend to avoid adding them into the tree.

There's already some degree of skipping particularly for BPF compilation/loading if the regeneration wasn't triggered based on the need for that (via `datapathRegenCtxt`), otherwise in general this seems to just be the regular configuration path. So it seems like there's some potential for early exit here, but yeah I do think that should probably be investigated subsequently as part of a dedicated PR.

Reply:
Agree, TODO's in source tend to be forgotten, removed. A dedicated PR should be brought forward once someone has a concrete idea what to do about this if anything.